packages/manager/apps/telecom/src/app/telecom/pack/resiliation/pack-resiliation.controller.js

```diff
@@ -206,163 +221,168 @@ export default /* @ngInject */ function PackResiliationCtrl(
-  };
+  updateAllInfluencedByCheckedSubServices(serviceType) {
+    this.checkSwitchState(serviceType);
+    this.updateFeeSummary();
+  }
 
   /**
    * True if serviceType has at least one sub service allowed to be kept,
    * false otherwise.
    */
-  this.hasKeepableSubServices = function hasKeepableSubServices(serviceType) {
-    return some(self.subServicesTerms[serviceType], (service) =>
+  hasKeepableSubServices(serviceType) {
+    return some(this.subServicesTerms[serviceType], (service) =>
       get(service, 'keepServiceTerms.isAllowed'),
     );
-  };
+  }
 
   /**
    * Compute the new price according to the new date
    * @returns {*}
    */
-  this.computePrice = function computePrice() {
-    self.computingPrice = true;
-    return OvhApiPackXdslResiliation.v6()
+  computePrice() {
+    this.computingPrice = true;
+    return this.OvhApiPackXdslResiliation.v6()
       .resiliationTerms(
         {
-          packName: $stateParams.packName,
-          resiliationDate: self.model.when
-            ? self.model.when.toISOString()
+          packName: this.$stateParams.packName,
+          resiliationDate: this.model.when
+            ? this.model.when.toISOString()
             : null,
         },
         null,
       )
       .$promise.then(
         (data) => {
-          self.resiliationTerms.data.due = data.due;
-          self.updateFeeSummary();
+          this.resiliationTerms.data.due = data.due;
+          this.updateFeeSummary();
         },
-        (err) => new TucToastError(err),
+        (err) => new this.TucToastError(err),
       )
       .finally(() => {
-        self.computingPrice = false;
+        this.computingPrice = false;
       });
-  };
+  }
 
   /**
    * Open the date picker
    * @param event
    */
-  this.openDatePicker = function openDatePicker(event) {
-    self.pickerOpened = true;
-    self.pickerOpenedPreventConflict = true;
+  openDatePicker(event) {
+    this.pickerOpened = true;
+    this.pickerOpenedPreventConflict = true;
     event.stopPropagation();
 
-    $timeout(() => {
-      self.pickerOpenedPreventConflict = false;
+    this.$timeout(() => {
+      this.pickerOpenedPreventConflict = false;
     }, 500);
-  };
+  }
 
   /**
    * Switch the date picker state, if is open then close,
    * if is closed then open it
    *
    * @param event
    */
-  this.switchDatePickerState = function switchDatePickerState(event) {
-    if (!self.pickerOpenedPreventConflict) {
-      self.pickerOpened = !self.pickerOpened;
+  switchDatePickerState(event) {
+    if (!this.pickerOpenedPreventConflict) {
+      this.pickerOpened = !this.pickerOpened;
     }
 
     event.stopPropagation();
-  };
+  }
 
   /**
    * Resiliate a pack
    * @param  {Object} pack   Pack to resiliate
    * @param  {Object} survey Reason to resiliate
    * @param {Boolean} accept If true the resiliation must be done
    */
-  this.resiliatePack = function resiliatePack() {
+  resiliatePack(slf) {
+    const self = slf;
```

Comment:
**question:** Why is this required?

Reply:
I don't know why but it works like that. If I don't those 2 different values, it doesn't work. If you have a better practice, I'll take it.